src/gt4py/next/ffront/foast_passes/type_deduction.py

```diff
@@ -835,9 +837,20 @@ def _visit_astype(self, node: foast.Call, **kwargs) -> foast.Call:
                 f"Invalid call to `astype`. Second argument must be a scalar type, but got {new_type}.",
             )
 
-        return_type = with_altered_scalar_kind(
-            value.type, getattr(ts.ScalarKind, new_type.id.upper())
-        )
+        if isinstance(value, foast.TupleExpr):
+            element_types_new = []
+            for element in value.elts:
```

Comment:
I would add an extra check here to make sure that `new_type.id` is a ScalarKind like it is done above in lines 832-838

Reply:
As the code in lines 832-838 already includes the check, wouldn't an extra check here make it redundant?

Reply:
yes, the check should be over `element.type`. Sorry